lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,522 @@
+        expect(dropdownElement).toHaveAttribute('aria-hidden', 'false');
+      });
+
+  it('calls onChange whenver the input\'s onChange event fires', async function() {
+    const onChange = jest.fn(),
+        inputElement = quickRender({ onChange }).getByRole('combobox');
+
+    expect(onChange).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, 'a');
+    expect(onChange).toHaveBeenCalledWith('a');
+  });
+
+  it('does not call onSearch whenever the input\'s onChange event fires with a value that does not differ' +
+    'after trimming', async function() {
+    const onSearch = jest.fn(),
+        { getByRole } = quickRender({ value: 'foo', onSearch }),
+        inputElement = getByRole('combobox');
+
+    await userEvent.type(inputElement, ' ');
+    expect(onSearch).not.toHaveBeenCalled();
+  });
+
+  it('calls onSearch whenever the input\'s onChange event fires with a value that differs after trimming, ' +
+    'passing the trimmed value', async function() {
+    const onSearch = jest.fn(),
+        { getByRole } = quickRender({ value: 'foo', onSearch }),
+        inputElement = getByRole('combobox');
+
+    await userEvent.type(inputElement, 'f');
+    expect(onSearch).toHaveBeenCalledWith('foof');
+  });
+
+  it('passes the disabled prop to the input and buttons', function() {
+    const { getByRole, rerender } = quickRender(),
+        inputElement = getByRole('combobox');
+
+    expect(inputElement).not.toBeDisabled();
+
+    rerender(<NxCombobox { ...minimalProps } disabled={undefined}/>);
+    expect(inputElement).not.toBeDisabled();
+
+    rerender(<NxCombobox { ...minimalProps } disabled={false}/>);
+    expect(inputElement).not.toBeDisabled();
+
+    rerender(<NxCombobox { ...minimalProps } disabled={true}/>);
+    expect(inputElement).toBeDisabled();
+
+    rerender(<NxCombobox { ...minimalProps } matches={ [{ id: '1', displayName: 'Foo' }] } disabled={false}/>);
+    expect(inputElement).not.toBeDisabled();
+
+    rerender(<NxCombobox { ...minimalProps } matches={ [{ id: '1', displayName: 'Foo' }] } disabled={true}/>);
+    expect(inputElement).toBeDisabled();
+  });
+
+  it('sets the listbox role on the dropdown', function() {
+    const { getByRole } = quickRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toBeInTheDocument();
+  });
+
+  it('sets an id on the dropdown and references it in the combobox input aria-controls', function() {
+    const { getByRole } = quickRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+        inputElement = getByRole('combobox'),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveAttribute('id');
+    expect(inputElement).toHaveAttribute('aria-controls', dropdownElement.id);
+  });
+
+  it('sets the alert role on the alert dropdown when it is in loading, error, or empty states', function() {
```

Comment:
For this test to really prove anything, it should also test that the `alert` is not present when not in one of the three listed states

Reply:
Updated here 6dbc601488edec3acdf06ca1c289688de300a1a8.